packages/babel-parser/src/parser/expression.js

```diff
@@ -1193,6 +1193,36 @@ export default class ExpressionParser extends LValParser {
         return this.parseTopicReferenceThenEqualsSign(tt.bitwiseXOR, "^");
       }
 
+      case tt.doubleCaret: {
+        const pipeProposal = this.getPluginOption(
+          "pipelineOperator",
+          "proposal",
+        );
+        const pluginTopicToken = this.getPluginOption(
+          "pipelineOperator",
+          "topicToken",
+        );
+
+        // The `^^` token is valid only when:
+        // the pipe-operator proposal is active,
+        // its "pipeProposal" is configured as "hack",
+        // and "topicToken" is configured as "^^".
+        // If the pipe-operator proposal settles on a token that is not ^^,
+        // then this token type may be removed.
+        if (pipeProposal === "hack" && pluginTopicToken === "^^") {
+          // `^^^` is forbidden and must be separated by a space.
```

Comment:
Since this is space-sensitive, I'd prefer to handle it in the tokenizer.

Reply:
Resolved in c9584de5a5d97f1c0d94f2a2d1c304bc0b24cefd.